common/src/main/java/com/redhat/cloud/notifications/routers/sources/SecretUtils.java

```diff
@@ -269,10 +285,10 @@ private long createBasicAuthentication(final BasicAuthentication basicAuthentica
      * @param orgId the organization id related to this operation for the tenant identification.
      * @return the id of the created secret.
      */
-    private long createSecretTokenSecret(final String secretToken, final String orgId) {
+    private long createSecretTokenSecret(final String secretToken, final String tokenType, final String orgId) {
         Secret secret = new Secret();
 
-        secret.authenticationType = Secret.TYPE_SECRET_TOKEN;
+        secret.authenticationType = tokenType;
         secret.password = secretToken;
```

Comment:
Hhmmm... I was thinking that maybe it makes more sense to remove both the `SecretUtils#createBasicAuthentication(final BasicAuthentication basicAuthentication, final String orgId)` and the `createSecretTokenSecret(final String secretToken, final String tokenType, final String orgId)` functions and simply create the following one:

```java
    /**
     * Creates a secret in Sources.
     * @param authenticationType the type of secret we are storing.
     * @param username the username of the secret.
     * @param password the password of the secret.
     * @param orgId the organization ID which this secret belongs to.
     * @return the created secret's ID.
     */
    private long createSecret(final String authenticationType, final String username, final String password, final String orgId) {
        final Secret secret = new Secret();

        secret.authenticationType = authenticationType;
        secret.password = password;
        secret.username = username;

        final Secret createdSecret = this.sourcesService.create(
            orgId,
            this.sourcesPsk,
            secret
        );

        return createdSecret.id;
    }
```

What do you think?

Reply:
For me it's already confusing to have to send a Bearer or a token on a field named `password` ;)
This is why I prefer havind dinstinct method to build the right kind of secret.

Reply:
Yeah, I have to admit it is a bit weird... Good, the only reason I was suggesting a change is because we are reusing the `createSecretTokenSecret` name, but for now is okay